### PR TITLE
[15.0][MIG] product_multi_barcode: Migration to 15.0

### PR DESCRIPTION
**Pull request by vib-adhoc**

### CODE REVIEW

**Review by nicomacr — APPROVED** [review]
LGTM

**Review by StefanRijnhart — APPROVED** [review]

File: product_multi_barcode/models/product_barcode.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        for rec in self.filtered(
+            lambda x: not x.product_id and x.product_tmpl_id.product_variant_ids
+        ):
+            rec.product_id = rec.product_tmpl_id.product_variant_ids[0]
+
+    @api.constrains("name")
+    def _check_duplicates(self):
+        for record in self:
+            barcodes = self.search(
+                [("id", "!=", record.id), ("name", "=", record.name)]
```
Comment (hieulucky111) [inline product_multi_barcode/models/product_barcode.py:55]: should use `search_count`?

File: product_multi_barcode/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            product.barcode = product.barcode_ids[:1].name
+
+    def _inverse_barcode(self):
+        for product in self:
+            if product.barcode_ids:
+                product.barcode_ids[:1].write({"name": product.barcode})
+            elif not product.barcode:
+                product.barcode_ids.unlink()
+            else:
+                self.env["product.barcode"].create(self._prepare_barcode_vals())
```
Comment (hieulucky111) [inline product_multi_barcode/models/product_product.py]: should be `product._prepare_barcode_vals()`

**Review by hieulucky111 — CHANGES_REQUESTED** [review]

File: product_multi_barcode/models/product_product.py (hunk 2 above)
Comment (vib-adhoc) [reply product_multi_barcode/models/product_product.py]: Thanks for your suggestion!

**Review by hieulucky111 — APPROVED** [review]
LGTM
Code review

**Comment by pere-aquarian** [comment]
Hi @vib-adhoc , 

Do you need any help here? 

Thank you!

**Comment by vib-adhoc** [comment]
Hi, @pere-aquarian 
I think it is ready to merge, if you want to fix de codecov check you can do it
Thank you for offering your help!

**Comment by pere-aquarian** [comment]
> Hi, @pere-aquarian I think it is ready to merge, if you want to fix de codecov check you can do it Thank you for offering your help!

I'll take a look asap!
thanks @vib-adhoc

File: product_multi_barcode/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    )
+
+    @api.depends("barcode_ids.name", "barcode_ids.sequence")
+    def _compute_barcode(self):
+        for product in self:
+            product.barcode = product.barcode_ids[:1].name
+
+    def _inverse_barcode(self):
+        for product in self:
+            if product.barcode_ids:
```
Comment (pere-aquarian) [inline product_multi_barcode/models/product_product.py:33]: @vib-adhoc , maybe the comparison must be more specific including [:1]. The same for the elif...

**Review by leemannd — APPROVED** [review]

File: product_multi_barcode/models/product_product.py (hunk 3 above)
Comment (StefanRijnhart) [reply product_multi_barcode/models/product_product.py:33]: That's not going to make any difference I think.

**Pull request merged**